### PR TITLE
[Feature] Pool opportunity length field

### LINKED ISSUE #8824: ✨ Pool: Length Of Opportunity
## ✨ Feature

We are going to be adding a new field for "Length Of Opportunity" for pools.

## 🕵️ Details

This new field should be a `<select>` and have the following `<options>`:
* Term (6 months)
* Term (1 year)
* Term (2 year)
* Indeterminate (Permanent)
* Various

## 🎨 Design File

@substrae 

Pool editor:
https://www.figma.com/file/e2DTvyppj41m6slFWTDZ0U/Admin---Job-advertisement-creation?type=design&node-id=52-40341&mode=design&t=cs4ucOAt8RuZ44or-0

Pool advertisement:
https://www.figma.com/file/CdYmWlP6SpT17bw7Q9411Z/Candidates---Job-advertisement?type=design&node-id=260-5309&mode=design&t=83ZYhqDekaDoFFyC-0

## 📸 Screenshot

Pool editor:
image
image

Pool advertisement
image
Link to external site: https://www.tpsgc-pwgsc.gc.ca/remuneration-compensation/collectivite-community/employeur-employer/emplfpf-emplfps-eng.html#a8

## ✅ Acceptance Criteria

- [ ] migration, database field, Eloquent model (Existing pools in the database use the Various option)
- [ ] seeder, factor updates
- [ ] Graphql field, enum, mutation update
- [ ] Pool editor, editable state
- [ ] Pool editor, readonly state
- [ ] Pool advertisement, card, with link to external site (follow established pattern)
- [ ] frontend "pool can publish" validation update
- [ ] backend "pool can publish" validation update

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #8824 

## 👋 Introduction

Adds a new opportunity length field to the pools table to allow admins to communicate the employment duration.

## 🧪 Testing

> [!IMPORTANT]
> Do not run a fresh migration while testing this. We want to test updating existing records!

### Migration/Seeders

1. Run migrations `docker-compose exec -w /home/site/wwwroot/api webserver sh -c "php artisan migrate"`
2. Confirm `pools` table has new column and existing rows are set to `VARIOUS`
3. Seed new data `docker-compose exec -w /home/site/wwwroot/api webserver sh -c "php artisan db:seed"`
4. Confirm new data has a variation of the enum values

### Publishing a Pool

1. Build app `npm run dev`
2. Login as admin
3. Create a new pool
4. Fill out the pool fields **leaving opportunity length empty**
5. Confirm the pool is still not complete and you cannot publish
6. Select a pool opportunity length
7. Confirm pool is now complete and you can publish

### Viewing a pool

1. View the advertisement on the public facing page
2. Confirm the selected opportunity appears
3. Confirm clicking the info link shows the appropriate page on both English and French

## 📸 Screenshot

Screenshot 2024-02-16 145126
Screenshot 2024-02-16 145143
Screenshot 2024-02-16 145155

### CODE REVIEW

**Review by vd1992 — CHANGES_REQUESTED** [review]
Looking good, just a few small things

File: apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx:1]: Oh right! Fixed in [fix edit pool basic info null/error state](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9395/commits/d7568568eb1c6d7116fa895e4ac7f853cf616da2)

File: apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx [hunk 1]
```diff
@@ -559,6 +564,37 @@ export const PoolPoster = ({
                   value={workLocation}
                   suffix={<WorkLocationDialog workLocation={workLocation} />}
                 />
+                <DataRow
+                  label={
+                    intl.formatMessage({
+                      defaultMessage: "Employment length",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx]: Oops! At least I was close 😅 Moved in [move length of opportunity field above work location](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9395/commits/cb58c8aa39fbc5ea42b53a693018bdd7023884b5)

**Review by vd1992 — APPROVED** [review]
Looks good to me, nice job!

**Pull request merged**